keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -2689,90 +2747,74 @@ previous answers based on experience in the field.
-  - 99% percentile over day of absolute value from (job creation time minus expected
-    job creation time) for cron job <= 10%
-  - 99.9% of /health requests per day finish with 200 code
-
-These goals will help you determine what you need to measure (SLIs) in the next
-question.
--->
+For Pods not using ResourceClaims, the same SLOs apply as before.
+
+For kube-controller-manager, metrics for the new controller could be checked to
+ensure that work items do not remain in the queue for too long, for some
+definition of "too long".
+
+Pod scheduling and startup are more important. However, expected performance
+will depend on how resources are used (for example, how often new Pods are
+created), therefore it is impossible to predict what reasonable SLOs might be.
+
+The resource manager component will do its work similarly to the
+existing volume manager, but the overhead and complexity should
+be lower:
+
+* Resource preparation should be fairly quick as in most cases it simply
+  creates CDI file 1-3 Kb in size. Unpreparing resource usually means
+  deleting CDI file, so it should be quick as well.
+
+* The complexity is lower than in the volume manager
+  because there is only one global operation needed (prepare vs.
+  attach + publish for each pod).
+
+* Reconstruction after a kubelet restart is simpler (call
+  NodePrepareResource again vs. trying to determine whether
+  volumes are mounted).
 
 ###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
 
-<!--
-Pick one more of these and delete the rest.
--->
+For kube-controller-manager, metrics similar to the generic ephemeral volume
+controller [were added](https://github.com/kubernetes/kubernetes/blob/163553bbe0a6746e7719380e187085cf5441dfde/pkg/controller/resourceclaim/metrics/metrics.go#L32-L47):
+
+- [X] Metrics
+  - Metric name: `resource_controller_create_total`
+  - Metric name: `resource_controller_create_failures_total`
+  - Metric name: `workqueue` with `name="resource_claim"`
 
-- [ ] Metrics
-  - Metric name:
-  - [Optional] Aggregation method:
-  - Components exposing the metric:
-- [ ] Other (treat as last resort)
-  - Details:
+For kube-scheduler and kubelet, existing metrics for handling Pods already
+cover most aspects. For example, in the scheduler the
+["unschedulable_pods"](https://github.com/kubernetes/kubernetes/blob/6f5fa2eb2f4dc731243b00f7e781e95589b5621f/pkg/scheduler/metrics/metrics.go#L200-L206)
+metric will call out pods that are currently unschedulable because of the
+`DynamicResources` plugin.
 
 ###### Are there any missing metrics that would be useful to have to improve observability of this feature?
 
-<!--
-Describe the metrics themselves and the reasons why they weren't added (e.g., cost,
-implementation difficulties, etc.).
--->
+No.
 
 ### Dependencies
 
-<!--
-This section must be completed when targeting beta to a release.
--->
+The container runtime must support CDI.
```

Comment:
I think that would be containerd 1.7.1 and all supported versions of CRIO. So this dependency should be satifisied.

Maybe its worth a comment as this is a pretty common item now.

Reply:
Not sure. For the sake of completeness I then would have to maintain a list of runtimes and which versions support CDI, even if it is just to proof a claim like "pretty common now". I don't think this belongs into the KEP.